write.go

```diff
@@ -621,7 +621,12 @@ func (wb *writeBuffer) writeRecord(attributes int8, baseTime time.Time, offset i
 	wb.writeVarInt(int64(milliseconds(timestampDelta)))
 	wb.writeVarInt(offsetDelta)
 
-	wb.writeVarBytes(msg.Key)
+	if msg.Key != nil {
+		wb.writeVarBytes(msg.Key)
```

Comment:
Should we handle this case in the `writeVarBytes` method instead? It could prevent future issues of the same kind.

Reply:
Good idea ;) I will do and test it ;)

Reply:
Change done and tested.
This also will fix the tombstone used for kafka with log compaction.